# CMU Movies Summary Corpus

- Authors: Zaynab, Lylia, Ali, Christian, Yassin

---

## Tasks

1. **Select Project & Initial Analyses**:
   1. Agree on a project proposal with team members.
   2. Perform initial analyses to verify feasibility of the proposed project, including any additional data.
   3. Acquaint yourself with the provided data, preprocess it, and perform descriptive statistics.

2. **Pipeline & Data Description**:
   1. Create a pipeline for data handling and preprocessing, documented in the notebook.
   2. Describe the relevant aspects of the data, including:
      1. Handling the size of the data.
      2. Understanding the data (formats, distributions, missing values, correlations, etc.).
      3. Considering data enrichment, filtering, and transformation according to project needs.
   3. Develop a plan for methods to be used, with essential mathematical details.
   4. Outline a plan for analysis and communication, discussing alternative approaches considered.

3. **GitHub Repository & Deliverables**:
   1. Create a public GitHub repository named `ada-2023-project-<team>` under the `epfl-ada` GitHub organization. ✅
   2. Ensure the repository contains:
      1. **README.md** file with:
         1. **Title**: Project title.
         2. **Abstract**: 150-word description of the project idea, goals, and motivation.
         3. **Research Questions**: List of research questions to address.
         4. **Proposed Additional Datasets**: Description of additional datasets, expected management, and feasibility analysis.
         5. **Methods**: Methods to be used in the project.
         6. **Proposed Timeline**: Timeline for the project.
         7. **Organization within the Team**: Internal milestones leading to Milestone P3.
         8. **Questions for TAs (optional)**: Any questions for the teaching assistants.
      2. **Code for Initial Analyses**: Structured code for initial analyses and data handling pipelines.
      3. **Notebook** presenting initial results, including:
         1. Main results and descriptive analysis.
         2. External scripts/modules for implementing core logic, to be called from the notebook.

---


## Table of Contents
- [1. Zaynab's part](##Zaynab's-part)
- [2. Lylia's part](##Lylia's-part)
- [3. Ali's part](##Ali's-part)
- [4. Cristians's part](##Christian's-part)
- [5. Yassin's part](##Yassin's-part)

---

### Library importation

In [10]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns

### data importation

In [11]:
DATA_PATH='./data/MovieSummaries/'

### movie metadata

In [12]:
movie_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'MovieName', 'ReleaseDate', 
    'BoxOfficeRevenue', 'Runtime', 'Languages', 'Countries', 'Genres'
]


movie_metadata = pd.read_csv(DATA_PATH+'movie.metadata.tsv', sep='\t', names=movie_columns)

movie_metadata

WikipediaMovieID FreebaseMovieID  \
0                975900       /m/03vyhn   
1               3196793       /m/08yl5d   
2              28463795      /m/0crgdbh   
3               9363483      /m/0285_cd   
4                261236       /m/01mrr1   
...                 ...             ...   
81736          35228177      /m/0j7hxnt   
81737          34980460      /m/0g4pl34   
81738           9971909      /m/02pygw1   
81739            913762       /m/03pcrp   
81740          12476867      /m/02w7zz8   

                                               MovieName ReleaseDate  \
0                                         Ghosts of Mars  2001-08-24   
1      Getting Away with Murder: The JonBenét Ramsey ...  2000-02-16   
2                                            Brun bitter        1988   
3                                       White Of The Eye        1987   
4                                      A Woman in Flames        1983   
...                                                  ...         ...   
81736                           Mermaids: The Body Found  2011-03-19   
81737                                            Knuckle  2011-01-21   
81738                                  Another Nice Mess  1972-09-22   
81739  The Super Dimension Fortress Macross II: Lover...  1992-05-21   
81740                                            Spliced        2002   

       BoxOfficeRevenue  Runtime                           Languages  \
0            14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1                   NaN     95.0  {"/m/02h40lc": "English Language"}   
2                   NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3                   NaN    110.0  {"/m/02h40lc": "English Language"}   
4                   NaN    106.0   {"/m/04306rv": "German Language"}   
...                 ...      ...                                 ...   
81736               NaN    120.0  {"/m/02h40lc": "English Language"}   
81737               NaN     96.0  {"/m/02h40lc": "English Language"}   
81738               NaN     66.0  {"/m/02h40lc": "English Language"}   
81739               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               Countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  Genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

### character metadata

In [13]:
character_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'ReleaseDate', 'CharacterName',
    'ActorDOB', 'ActorGender', 'ActorHeight', 'ActorEthnicity', 
    'ActorName', 'ActorAgeAtRelease', 'FreebaseCharacterActorMapID',
    'FreebaseCharacterID', 'FreebaseActorID'
]

character_metadata = pd.read_csv(DATA_PATH+'character.metadata.tsv', sep='\t', names=character_columns)


character_metadata

WikipediaMovieID FreebaseMovieID ReleaseDate  \
0                 975900       /m/03vyhn  2001-08-24   
1                 975900       /m/03vyhn  2001-08-24   
2                 975900       /m/03vyhn  2001-08-24   
3                 975900       /m/03vyhn  2001-08-24   
4                 975900       /m/03vyhn  2001-08-24   
...                  ...             ...         ...   
450664            913762       /m/03pcrp  1992-05-21   
450665            913762       /m/03pcrp  1992-05-21   
450666          28308153      /m/0cp05t9        1957   
450667          28308153      /m/0cp05t9        1957   
450668          28308153      /m/0cp05t9        1957   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                      Elensh     1970-05           F          NaN   
450665                      Hibiki  1965-04-12           M          NaN   
450666                         NaN  1941-11-18           M        1.730   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN   

       ActorEthnicity           ActorName  ActorAgeAtRelease  \
0                 NaN      Wanda De Jesus               42.0   
1          /m/044038p  Natasha Henstridge               27.0   
2             /m/0x67            Ice Cube               32.0   
3                 NaN       Jason Statham               33.0   
4                 NaN         Clea DuVall               23.0   
...               ...                 ...                ...   
450664            NaN  Dorothy Elias-Fahn                NaN   
450665            NaN       Jonathan Fahn               27.0   
450666      /m/02w7gg      David Hemmings               15.0   
450667            NaN    Roberta Paterson                NaN   
450668            NaN         John Rogers                NaN   

       FreebaseCharacterActorMapID FreebaseCharacterID FreebaseActorID  
0                       /m/0bgchxw          /m/0bgcj3x      /m/03wcfv7  
1                        /m/0jys3m          /m/0bgchn4       /m/0346l4  
2                        /m/0jys3g          /m/0bgchn_      /m/01vw26l  
3                       /m/02vchl6          /m/0bgchnq       /m/034hyc  
4                       /m/02vbb3r          /m/0bgchp9       /m/01y9xg  
...                            ...                 ...             ...  
450664                  /m/0kr406c          /m/0kr406h       /m/0b_vcv  
450665                  /m/0kr405_          /m/0kr4090       /m/0bx7_j  
450666                  /m/0g8ngmc                 NaN       /m/022g44  
450667                  /m/0g8ngmj                 NaN      /m/0g8ngmm  
450668                  /m/0g8ngmw                 NaN      /m/0btz19d  

[450669 rows x 13 columns]

### plot summaries

In [14]:
plot_columns = ['WikipediaMovieID', 'PlotSummary']

plot_summaries = pd.read_csv(DATA_PATH+'plot_summaries.txt', sep='\t', names=plot_columns)

plot_summaries


WikipediaMovieID                                        PlotSummary
0              23890098  Shlykov, a hard-working taxi driver and Lyosha...
1              31186339  The nation of Panem consists of a wealthy Capi...
2              20663735  Poovalli Induchoodan  is sentenced for six yea...
3               2231378  The Lemon Drop Kid , a New York City swindler,...
4                595909  Seventh-day Adventist Church pastor Michael Ch...
...                 ...                                                ...
42298          34808485  The story is about Reema , a young Muslim scho...
42299           1096473  In 1928 Hollywood, director Leo Andreyev  look...
42300          35102018  American Luthier focuses on Randy Parsons’ tra...
42301           8628195  Abdur Rehman Khan , a middle-aged dry fruit se...
42302           6040782  1940 - Operation Dynamo has just taken place. ...

[42303 rows x 2 columns]

### name clusters

In [15]:
name_clusters_columns = ['FreebaseCharacterActorMapID', 'CharacterName']

name_clusters = pd.read_csv(DATA_PATH+'name.clusters.txt', sep='\t', names=name_clusters_columns)

name_clusters


FreebaseCharacterActorMapID CharacterName
0                  Stuart Little     /m/0k3w9c
1                  Stuart Little     /m/0k3wcx
2                  Stuart Little     /m/0k3wbn
3                       John Doe     /m/0jyg35
4                       John Doe     /m/0k2_zn
...                          ...           ...
2661                  John Rolfe     /m/0k5_ql
2662                  John Rolfe    /m/02vd6vs
2663             Elizabeth Swann     /m/0k1xvz
2664             Elizabeth Swann     /m/0k1x_d
2665             Elizabeth Swann     /m/0k1xxy

[2666 rows x 2 columns]

### TV tropes clusters

In [16]:
tvtropes_columns = ['FreebaseCharacterActorMapID', 'CharacterType']

tvtropes_clusters = pd.read_csv(DATA_PATH+'tvtropes.clusters.txt', sep='\t', names=tvtropes_columns)

tvtropes_clusters


FreebaseCharacterActorMapID  \
0       absent_minded_professor   
1       absent_minded_professor   
2       absent_minded_professor   
3       absent_minded_professor   
4       absent_minded_professor   
..                          ...   
496                   young_gun   
497                   young_gun   
498                   young_gun   
499                   young_gun   
500                   young_gun   

                                         CharacterType  
0    {"char": "Professor Philip Brainard", "movie":...  
1    {"char": "Professor Keenbean", "movie": "Richi...  
2    {"char": "Dr. Reinhardt Lane", "movie": "The S...  
3    {"char": "Dr. Harold Medford", "movie": "Them!...  
4    {"char": "Daniel Jackson", "movie": "Stargate"...  
..                                                 ...  
496  {"char": "Morgan Earp", "movie": "Tombstone", ...  
497  {"char": "Colorado Ryan", "movie": "Rio Bravo"...  
498  {"char": "Tom Sawyer", "movie": "The League of...  
499  {"char": "William H. 'Billy the Kid' Bonney", ...  
500  {"char": "Jake", "movie": "Silverado", "id": "...  

[501 rows x 2 columns]

---

## Zaynab's part

---
## Lylia's part

In [84]:
# Load the dataset
box_office=pd.read_csv('/Users/lylou/Downloads/BoxOfficeMojo/boxoffice.csv')
box_office.head(25)

rank                                       title studio  lifetime_gross  \
0      1                Star Wars: The Force Awakens     BV       936662225   
1      2                                      Avatar    Fox       760507625   
2      3                               Black Panther     BV       700059566   
3      4                      Avengers: Infinity War     BV       678681680   
4      5                                     Titanic   Par.       659363944   
5      6                              Jurassic World   Uni.       652270625   
6      7                       Marvel's The Avengers     BV       623357910   
7      8                    Star Wars: The Last Jedi     BV       620181382   
8      9                               Incredibles 2     BV       597066966   
9     10                             The Dark Knight     WB       534858444   
10    11                Rogue One: A Star Wars Story     BV       532177324   
11    12                 Beauty and the Beast (2017)     BV       504014165   
12    13                                Finding Dory     BV       486295561   
13    14   Star Wars: Episode I - The Phantom Menace    Fox       474544677   
14    15                                   Star Wars    Fox       460998007   
15    16                     Avengers: Age of Ultron     BV       459005868   
16    17                       The Dark Knight Rises     WB       448139099   
17    18                                     Shrek 2     DW       441226247   
18    19                 E.T.: The Extra-Terrestrial   Uni.       435110554   
19    20             The Hunger Games: Catching Fire    LGF       424668047   
20    21  Pirates of the Caribbean: Dead Man's Chest     BV       423315812   
21    22                               The Lion King     BV       422783777   
22    23                                 Toy Story 3     BV       415004880   
23    24              Jurassic World: Fallen Kingdom   Uni.       413025695   
24    25                                Wonder Woman     WB       412563408   

    year  
0   2015  
1   2009  
2   2018  
3   2018  
4   1997  
5   2015  
6   2012  
7   2017  
8   2018  
9   2008  
10  2016  
11  2017  
12  2016  
13  1999  
14  1977  
15  2015  
16  2012  
17  2004  
18  1982  
19  2013  
20  2006  
21  1994  
22  2010  
23  2018  
24  2017

In [59]:
box_office.shape

(15743, 5)

Lifetime gross = total box office revenue over the entire time the movie runs in theater 
The dataset is ordered by the most successful (highest boxoffice rev) movies.

In [67]:
movie_metadata.shape

(81741, 10)

In [46]:
#Check for NaNs
missing_values = box_office.isnull().sum()
missing_values

rank               0
title              0
studio            35
lifetime_gross     0
year               0
dtype: int64

In [85]:
missing_values = box_office.isna().sum()
missing_values


rank               0
title              0
studio            35
lifetime_gross     0
year               0
dtype: int64

In [ ]:
# Check the min and max date in movie metadata
# Convert 'Release Data' to datetime format
movie_metadata['ReleaseDate'] = pd.to_datetime(movie_metadata['ReleaseDate'], errors='coerce')

# Extract the year from the 'Release Data' column
movie_metadata['year'] = movie_metadata['ReleaseDate'].dt.year

# Find the min and max year
min_year = movie_metadata['year'].min()
max_year = movie_metadata['year'].max()

#movie_metadata.head(2)
min_year, max_year


(1892.0, 2016.0)

In [65]:
# Remove the movies that are before 1888 and after 2016 as they arent going to be in movie_metadata
box_office_filtered = box_office[(box_office['year'] >= 1888) & (box_office['year'] <= 2016)]

# Size after filetring 
box_office_filtered.shape

(14585, 5)

In [ ]:
# How much info did we loose in boxoffice (not relevant)
loss = 100-(box_office_filtered.shape[0]*100)/box_office.shape[0]
print('We lost',loss,'% of boxoffice df after filtering')

We lost 7.355650130216603 % of boxoffice df after filtering


In [69]:
# Merge movie_metadata and box_office_filtered on movie titles (outer merge)
movie_merged = pd.merge(box_office_filtered, movie_metadata, how='right', left_on='title', right_on='MovieName')

# Check the number of rows after the merge to check if it was successful
movie_merged.shape


(81841, 15)

In [74]:
# We have 100 more rows than before i.e some movies in boxoffice are duplicated
duplicates_in_box_office = box_office_filtered[box_office.duplicated(subset='title', keep=False)]
duplicates_in_box_office.shape


/var/folders/pq/vs1ny6651qz1x3k8m4bz09m40000gn/T/ipykernel_9183/4146444921.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  duplicates_in_box_office = box_office_filtered[box_office.duplicated(subset='title', keep=False)]


(80, 5)

Doesnt make sense we should have 100 not 80. Maybe triplicates?

In [83]:
# We remove duplicates 
box_office_cleaned = box_office_filtered.drop_duplicates(subset='title', keep='first')

# New merge with the cleaned box_office_filtered 
movie_merged_cleaned = pd.merge(movie_metadata, box_office_cleaned, how='left', left_on='MovieName', right_on='title')

movie_merged_cleaned.shape


(81741, 15)

Removing the duplicates was enough to go back to the original shape

In [78]:
movie_merged_cleaned.head(5)

WikipediaMovieID FreebaseMovieID  \
0            975900       /m/03vyhn   
1           3196793       /m/08yl5d   
2          28463795      /m/0crgdbh   
3           9363483      /m/0285_cd   
4            261236       /m/01mrr1   

                                           MovieName ReleaseDate  \
0                                     Ghosts of Mars  2001-08-24   
1  Getting Away with Murder: The JonBenét Ramsey ...  2000-02-16   
2                                        Brun bitter         NaT   
3                                   White Of The Eye         NaT   
4                                  A Woman in Flames         NaT   

   BoxOfficeRevenue  Runtime                           Languages  \
0        14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1               NaN     95.0  {"/m/02h40lc": "English Language"}   
2               NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3               NaN    110.0  {"/m/02h40lc": "English Language"}   
4               NaN    106.0   {"/m/04306rv": "German Language"}   

                                   Countries  \
0  {"/m/09c7w0": "United States of America"}   
1  {"/m/09c7w0": "United States of America"}   
2                     {"/m/05b4w": "Norway"}   
3             {"/m/07ssc": "United Kingdom"}   
4                    {"/m/0345h": "Germany"}   

                                              Genres  year_x  rank title  \
0  {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   NaN   NaN   
1  {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  2000.0   NaN   NaN   
2  {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...     NaN   NaN   NaN   
3  {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...     NaN   NaN   NaN   
4                            {"/m/07s9rl0": "Drama"}     NaN   NaN   NaN   

  studio  lifetime_gross  year_y  
0    NaN             NaN     NaN  
1    NaN             NaN     NaN  
2    NaN             NaN     NaN  
3    NaN             NaN     NaN  
4    NaN             NaN     NaN

In [79]:
# We check for the NaNs in the merged dataset i.e the movies in metadata that don't have a ranking or a lifetime_gross
nan_check = movie_merged_cleaned[['lifetime_gross', 'rank']].isna().sum()

# Display the number of NaNs in each of these columns
nan_check


lifetime_gross    70293
rank              70293
dtype: int64

70293 out of 81741 movies don't have a ranking or a lifetime_gross. **Only 14% of the movies have these info.**

---
## Ali's part

---
## Christian's part

---
## Yassin's part